examples/general/non_local_effect_before_error_return/src/lib.rs

```diff
@@ -150,7 +150,7 @@ impl<'tcx> LateLintPass<'tcx> for NonLocalEffectBeforeErrorReturn {
             return;
         }
 
-        if !is_result(cx, cx.typeck_results().expr_ty(body.value)) {
+        if !is_permitted_result(cx, cx.typeck_results().expr_ty(body.value)) {
```

Comment:
```suggestion
        if !is_disallowed_result(cx, cx.typeck_results().expr_ty(body.value)) {
```
I know I suggested `is_permitted_result` in #929, but I see now that it doesn't make sense.

We want to continue if type of the body is a result that is _not_ permitted.

So I am suggesting now `is_disallowed result`, but I don't love it.

Do you have a name you think is more apt?

Reply:
@smoelius I was thinking something along the lines of `is_lintable_result` or `is_lint_target`. I came up  with these due to the fact that it's clear that we're determining whether to apply the lint or not in this fn. What do you think about it?
For now i'm making changes with `is_disallowed_result` just as a placeholder, let me  know your choice, I'll make another commit to update it.

Reply:
I like `is_lintable_result`. Let's please go with that. :)